modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -372,53 +362,70 @@ private CompletableFuture<AsyncSqlCursor<List<Object>>> querySingle0(
 
         CompletableFuture<AsyncSqlCursor<List<Object>>> stage = start
                 .thenApply(v -> {
-                    var nodes = Commons.parse(sql, FRAMEWORK_CONFIG.getParserConfig());
+                    var nodes = Commons.parse(sql, Commons.PARSER_CONFIG);
 
                     if (nodes.size() > 1) {
                         throw new SqlException(QUERY_INVALID_ERR, "Multiple statements aren't allowed.");
                     }
 
                     return nodes.get(0);
                 })
-                .thenCompose(sqlNode -> prepareSvc.prepareAsync(sqlNode, ctx))
-                .thenApply(plan -> {
-                    context.maybeUnwrap(QueryValidator.class)
-                            .ifPresent(queryValidator -> queryValidator.validatePlan(plan));
-
-                    // Transactional DDL is not supported as well as RO transactions, hence
-                    // only DML requiring RW transaction is covered
-                    boolean implicitTxRequired = plan.type() == Type.DML && outerTx == null;
-                    InternalTransaction implicitTx = implicitTxRequired ? txManager.begin() : null;
-
-                    BaseQueryContext enrichedContext = implicitTxRequired ? ctx.toBuilder().transaction(implicitTx).build() : ctx;
-
-                    var dataCursor = executionSrvc.executePlan(plan, enrichedContext);
-
-                    return new AsyncSqlCursorImpl<>(
-                            SqlQueryType.mapPlanTypeToSqlType(plan.type()),
-                            plan.metadata(),
-                            implicitTx,
-                            new AsyncCursor<List<Object>>() {
-                                @Override
-                                public CompletionStage<BatchedResult<List<Object>>> requestNextAsync(int rows) {
-                                    session.touch();
-
-                                    return dataCursor.requestNextAsync(rows);
-                                }
-
-                                @Override
-                                public CompletableFuture<Void> closeAsync() {
-                                    session.touch();
-
-                                    return dataCursor.closeAsync();
-                                }
-                            }
-                    );
+                .thenCompose(sqlNode -> {
+                    BaseQueryContext ctx = BaseQueryContext.builder()
+                            .cancel(new QueryCancel())
```

Comment:
It looks strange to use new QueryCncel object here

Reply:
Good catch! Fixed